### ISSUE TITLE
Is it possible to check if dll is netframework/netcore/netstandard using MetadataReader?

### ISSUE THREAD

**Issue by eddynaka** [dotnet/runtime#48699]
Hi,

today I have the following code that is trying to identify if a dll is netframework/netcore/netstandard:
```csharp
internal static ManagedPlatform ComputeIsDotNetCore(MetadataReader metadataReader)
{
    foreach (AssemblyReferenceHandle handle in metadataReader.AssemblyReferences)
    {
        AssemblyReference assemblyReference = metadataReader.GetAssemblyReference(handle);
        StringHandle stringHandle = assemblyReference.Name;
        string assemblyName = metadataReader.GetString(stringHandle);

        switch (assemblyName)
        {
            case "mscorlib":
            {
                return ManagedPlatform.DotNetFramework;
            }

            case "System.Runtime":
            {
                return ManagedPlatform.DotNetCore;
            }

            case "netstandard":
            {
                return ManagedPlatform.DotNetStandard;
            }

            default:
            {
                break;
            }
        }
    }

    throw new InvalidOperationException("Could not identify managed platform.");
}
```

For a few dlls generated in runtime linux, net5 and in obj/linked folder, i can't identify what type is it.

Attached you can find a dll that was generated with those flags and I can't read anything that I could identify.
[linked.zip](https://github.com/dotnet/runtime/files/6036122/linked.zip)

**Comment by eddynaka** [comment]
@krwq , don't know if you know this or if you could point me to.
thank you!

**Comment by alexrp** [comment]
You probably need to handle `System.Private.CoreLib` for your `DotNetCore` case. I would imagine (without checking) that some platform assemblies reference it directly rather than `System.Runtime`.

**Comment by eddynaka** [comment]
Right. I saw that `CoreLib` was one of the hits, but I wasn't sure if that was a good check.

**Comment by kevingosse** [comment]
I believe the most reliable way would be to check the value of the `TargetFramework` attribute, but I found no straightforward way to decode the value of the attribute from the MetadataReader:

```csharp

foreach (var handle in metadata.CustomAttributes)
{
    var attribute = metadata.GetCustomAttribute(handle);
    var ctor = metadata.GetMemberReference((MemberReferenceHandle)attribute.Constructor);
    var type = metadata.GetTypeReference((TypeReferenceHandle)ctor.Parent);
    var typeName = metadata.GetString(type.Name);

    if (typeName == "TargetFrameworkAttribute")
    {
        var bytes = metadata.GetBlobBytes(attribute.Value);

        // ?
    }
}
```

**Comment by krwq** [comment]
Usually you should design your app so that you depend on the information in the nuget package rather than dll but having said that I think checking assembly references + attributes is probably the best approach you can get from just DLL. Unfortunately I do not have a specific guidelines I can share, probably easiest is to create couple of test assemblies and compare with ILSpy or IL DASM.

For custom attributes and metadata reader in general it's best to refer to the ECMA-335 spec and then search for corresponding API, i.e.: `II.21` mentions that, also you can take a look at MetadtaReader tests, i.e. https://github.com/dotnet/runtime/blob/master/src/libraries/System.Reflection.Metadata/tests/Metadata/Decoding/CustomAttributeDecoderTests.cs - if you know specific attribute you're looking for then it should be much simpler to decode

**Comment by eddynaka** [comment]
Looking again at this, there are cases where either my approach or @kevingosse does not work.
Also tried to open in ILSpy and DotPeek.
For ILSpy it is showing as NET4.0 and for DotPeek NET4.8.

executed the dumpbin to see if i can easily see something. but again, no clear information.
